book-a-reading-room-visit.api/Service/Impl/WorkingDayService.cs

```diff
@@ -103,5 +103,19 @@ public async Task<DateTime> GetNextBulkOrderOpeningDayAsync(DateTime dateTime, i
             }
             return dateToReturn;
         }
+
+        public async Task<DateTime> GetCompleteByDateAsync(DateTime dateTime)
+        {
+            int daysToDeduct = int.Parse(_configuration.GetSection("BookingTimeLine:CompleteBy").Value);
+            var holidays = await _bankHolidayAPI.GetBankHolidaysAsync();
+            var dateToReturn = dateTime;
+            while (daysToDeduct > 0)
+            {
+                dateToReturn = dateToReturn.AddDays(-1);
+                if (WorkingWeekDays.Contains(dateToReturn.DayOfWeek) && !holidays.Contains(dateToReturn))
+                    daysToDeduct--;
```

Comment:
Should dateToReturn = dateToReturn.AddDays(-1); not come after
                if (WorkingWeekDays.Contains(dateToReturn.DayOfWeek) && !holidays.Contains(dateToReturn)) ?

Reply:
@ACTNA it works, as the first date passed is already a working day so I deduct a day until I found the correct working day, so the main logic actually rely on the daysToDeduct counter.

Reply:
Only working days passed to GetCompleteByDateAsync.